.github/workflows/build-and-publish.yaml

```diff
@@ -0,0 +1,52 @@
+name: Check, build and publish
+
+on:
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  publish:
+    name: Publish to ghcr.io
+    runs-on: ubuntu-22.04
```

Comment:
Similar to my comment below on docker, I would just build on ubuntu and not caring about the version so that we don't really have strong arguments to rely on a particular version and thus won't need to perform a migration here.

Reply:
done!